instrumentation/apache-camel-2.20/src/test/groovy/test/MulticastDirectCamelTest.groovy

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test
+
+import static io.opentelemetry.trace.Span.Kind.INTERNAL
+
+import io.opentelemetry.instrumentation.test.AgentTestRunner
+import org.apache.camel.CamelContext
+import org.apache.camel.ProducerTemplate
+import org.springframework.boot.SpringApplication
+import org.springframework.context.ConfigurableApplicationContext
+import spock.lang.Shared
+
+class MulticastDirectCamelTest extends AgentTestRunner {
+
+  @Shared
+  ConfigurableApplicationContext server
+
+  def setupSpec() {
+    def app = new SpringApplication(MulticastConfig)
+    server = app.run()
+  }
+
+  def cleanupSpec() {
+    if (server != null) {
+      server.close()
+      server = null
+    }
+  }
+
+  def "parallel multicast to two child services"() {
+    setup:
+    def camelContext = server.getBean(CamelContext)
+    ProducerTemplate template = camelContext.createProducerTemplate()
+
+    when:
+    template.sendBody("direct:input", "Example request")
+
+    then:
+    assertTraces(1) {
+      trace(0, 3) {
+        def parent = it
+        it.span(0) {
+          name "input"
+          kind INTERNAL
+          hasNoParent()
+          attributes {
+            "camel.uri" "direct://input"
+          }
+        }
+        it.span(1) {
+          name "second"
```

Comment:
Are you sure that this is 100% deterministic?

Reply:
In all of the tests I performed it was, although I could not find a way to prove this synthetically.